### PR TITLE
[Bugfix] If there is no shuffle, we shouldn't acess rss

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
If there is no shuffle, we shouldn't acess rss, otherwise we will encounter an error

### Why are the changes needed?
If a application has no shuffle, it will occur a error

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manual test

### CODE REVIEW

File: client-mr/src/main/java/org/apache/hadoop/mapreduce/v2/app/RssMRAppMaster.java [hunk 1]
```diff
@@ -60,112 +60,114 @@ public static void main(String[] args) {
...
-        1, Sets.newHashSet(Constants.SHUFFLE_SERVER_VERSION));
-    Map<ShuffleServerInfo, List<PartitionRange>> serverToPartitionRanges = response.getServerToPartitionRanges();
-    final ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(
-        new ThreadFactory() {
-          @Override
-          public Thread newThread(Runnable r) {
-            Thread t = Executors.defaultThreadFactory().newThread(r);
-            t.setDaemon(true);
-            return t;
+    if (numReduceTasks > 0) {
```
Comment (frankliee) [inline client-mr/src/main/java/org/apache/hadoop/mapreduce/v2/app/RssMRAppMaster.java:63]: What  about numReduceTasks == 0 ???

Comment (jerqi) [reply client-mr/src/main/java/org/apache/hadoop/mapreduce/v2/app/RssMRAppMaster.java:63]: `numReduceTasks == 0` means no shuffle

Comment (frankliee) [reply client-mr/src/main/java/org/apache/hadoop/mapreduce/v2/app/RssMRAppMaster.java:63]: Which MR applications have no shuffle?

Comment (jerqi) [reply client-mr/src/main/java/org/apache/hadoop/mapreduce/v2/app/RssMRAppMaster.java:63]: map only, distcp

**Review by frankliee — APPROVED** [review]

**Pull request merged**